Either your suggested change, or my original patch, will build fine **with** the other two changes (conditional references) included.  Neither version will build without the changes to the references lines.

As far as I can see, the only difference with your suggestion is that it removes the possibility to override the CrossgenOutput setting manually on the command line.  I guess this doesn't really matter either way to me -- if you prefer the version with the s390x check on a separate line, I'm happy with that.


I'm not sure if you're still interested in the specific errors that occur when removing the changes to the references lines, but just in case, the first error is:
```
/home/uweigand/aspnetcore/eng/targets/ResolveReferences.targets(220,5): error MSB3245: Could not resolve this reference. Could not locate the package or project for "Microsoft.NETCore.App.Runtime.linux-s390x". Did you update baselines and dependencies lists? See docs/ReferenceResolution.md for more details. [/home/uweigand/aspnetcore/src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj]
```

This error goes away when adding the following to `eng/Dependencies.props`: 
```
    <LatestPackageReference Include="Microsoft.NETCore.App.Runtime.linux-s390x" />
```

and then I immediately get instead:
```
/home/uweigand/aspnetcore/eng/targets/ResolveReferences.targets(220,5): error MSB3245: Could not resolve this reference. Could not locate the package or project for "Microsoft.NETCore.App.Crossgen2.linux-s390x". Did you update baselines and dependencies lists? See docs/ReferenceResolution.md for more details. [/home/uweigand/aspnetcore/src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj]                                                                                                                                                
```

which I can similarly fix by adding to `eng/Dependencies.props`:
```
    <LatestPackageReference Include="Microsoft.NETCore.App.Crossgen2.linux-s390x" />
```

However, then I get the following errors:
```
/home/uweigand/aspnetcore/src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.linux-s390x. No packages exist with this id in source(s): dotnet-eng, dotnet-experimental, dotnet-public, dotnet-tools, dotnet31-transport, dotnet5, dotnet5-transport, dotnet6, dotnet6-transport, richnav [/home/uweigand/dotnet-net6/sdk/6.0.100-preview.7.21379.14/NuGet.targets]                                                                                                              
/home/uweigand/aspnetcore/src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Crossgen2.linux-s390x. No packages exist with this id in source(s): dotnet-eng, dotnet-experimental, dotnet-public, dotnet-tools, dotnet31-transport, dotnet5, dotnet5-transport, dotnet6, dotnet6-transport, richnav [/home/uweigand/dotnet-net6/sdk/6.0.100-preview.7.21379.14/NuGet.targets]                                                                                                            
```

The first of those I can fix by adding a local repo holding the `Microsoft.NETCore.App.Runtime.linux-s390x` package to `NuGet.config`.  But the second I cannot fix because the `Crossgen2.linux-s390x` package really does not exist anywhere.

Of course, making the References lines conditional as in the PR will fix both those issues.

--- a/src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj
+++ b/src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj
@@ -42,7 +42,8 @@ This package is an internal implementation of the .NET Core SDK and is not meant
     <!-- Pack .ni.r2rmap files in symbols package (native symbols for Linux) -->
     <AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder>$(AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder);.r2rmap</AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder>
 
-    <!-- Optimize the framework using the crossgen2 tool -->
+    <!-- Optimize the framework using the crossgen2 tool. Crossgen2 is not currently supported on s390x. -->
+    <CrossgenOutput Condition=" '$(TargetArchitecture)' == 's390x' ">false</CrossgenOutput>
     <CrossgenOutput Condition=" '$(CrossgenOutput)' == '' AND '$(Configuration)' != 'Debug' ">true</CrossgenOutput>
 
     <!-- Produce crossgen2 profiling symbols (.ni.pdb or .r2rmap files). -->
@@ -128,7 +129,8 @@ This package is an internal implementation of the .NET Core SDK and is not meant
     <Reference Include="System.Security.Permissions" ExcludeAssets="All" />
     <Reference Include="System.Windows.Extensions" ExcludeAssets="All" />
 
-    <Reference Include="Microsoft.NETCore.App.Runtime.$(RuntimeIdentifier)"
+    <Reference Condition="'$(CrossgenOutput)' == 'true'"
+        Include="Microsoft.NETCore.App.Runtime.$(RuntimeIdentifier)"
         ExcludeAssets="All"
         PrivateAssets="All"
         GeneratePathProperty="true" />
@@ -137,7 +139,8 @@ This package is an internal implementation of the .NET Core SDK and is not meant
       This package contains the crossgen2 tool. Unfortunately, it doesn't make the tool easy to use.
       $(GeneratePathProperty) and hacks in the _ExpandRuntimePackageRoot target work around the gaps.
     -->
-    <Reference Include="Microsoft.NETCore.App.Crossgen2.$(BuildOsName)-$(Crossgen2BuildArchitecture)"
+    <Reference Condition="'$(CrossgenOutput)' == 'true'"
+        Include="Microsoft.NETCore.App.Crossgen2.$(BuildOsName)-$(Crossgen2BuildArchitecture)"
         ExcludeAssets="All"
         PrivateAssets="All"
         GeneratePathProperty="true" />

